app/models/concerns/alchemy/picture_thumbnails.rb

```diff
@@ -150,6 +152,8 @@ def inferred_dimensions_from_string(string)
       width, height = dimensions_from_string(string)
       ratio = image_file_width.to_f / image_file_height.to_i
```

Comment:
Wouldn't the division by zero error raise here? 
How can `ratio.nan?` below ever be reached than? 
Shouldn't we return here if `image_file_height.nil?` instead?

Reply:
No, it doesn't raise, it returns `Float::NaN`. 
Yes, we can also return earlier if `image_file_height.nil?`. Would you prefer that? Happy to make the change.

Reply:
What version of Ruby are you using?

In Ruby 3.2.2 this is what happens for me

```rb
>> 1/0
(irb):1:in `/': divided by 0 (ZeroDivisionError)
>> 1.0/0
=> Infinity
>> 1.0.class
=> Float
```

When does it return `Float::NAN`/`NaN`?

Reply:
Ah

```rb
>> 0.0/0
=> NaN
```

Reply:
But this is not your case, isn't it?

Reply:
That's precisely the case; the big happens because there's no picture yet :)